Makefile

```diff
@@ -1792,12 +1792,45 @@ ensure-js-deps:
 	@if [[ "${WEBASSETS_SKIP_BUILD}" -eq 1 ]]; then mkdir -p webassets/teleport && touch webassets/teleport/index.html; \
 	else $(MAKE) ensure-js-package-manager && pnpm install --frozen-lockfile; fi
 
+.PHONY: ensure-wasm-deps
+ifeq ($(WEBASSETS_SKIP_BUILD),1)
+ensure-wasm-deps:
+else
+ensure-wasm-deps: ensure-wasm-bindgen ensure-wasm-pack
+
+# Get the version of wasm-bindgen from cargo, as that is what wasm-pack is
+# going to do when it checks for the right version. The buildboxes do not
+# have jq installed (yet), so have a hacky awk version on standby.
+CARGO_GET_VERSION_JQ = cargo metadata --locked --format-version=1 | jq -r 'first(.packages[] | select(.name? == "$(1)") | .version)'
+CARGO_GET_VERSION_AWK = awk -F '[ ="]+' '/^name = "$(1)"$$/ {inpkg = 1} inpkg && $$1 == "version" {print $$2; exit}' Cargo.lock
+
+BIN_JQ = $(shell which jq 2>/dev/null)
+CARGO_GET_VERSION = $(if $(BIN_JQ),$(CARGO_GET_VERSION_JQ),$(CARGO_GET_VERSION_AWK))
+
+ensure-wasm-bindgen: NEED_VERSION = $(shell $(call CARGO_GET_VERSION,wasm-bindgen))
+ensure-wasm-bindgen: INSTALLED_VERSION = $(lastword $(shell wasm-bindgen --version 2>/dev/null))
+ensure-wasm-bindgen:
+	@: $(or $(NEED_VERSION),$(error Unknown wasm-bindgen version. Is it in Cargo.lock?))
+	$(if $(filter-out $(INSTALLED_VERSION),$(NEED_VERSION)),\
+		cargo install wasm-bindgen-cli --locked --version "$(NEED_VERSION)", \
```

Comment:
Forcing the installation and/or reinstallation of `wasm-bindgen-cli` as an actual binary instead of relying on `wasm-pack`'s private cache is going to be really annoying (especially as we switch branches), the fact that we need `wasm-pack` installed globally is already annoying. Can we keep this as a CI-only thing?

Reply:
Gus has done this, but I'm not sure this can be CI-only. We can no longer let wasm-pack build wasm-bindgen-cli as it just doesn't build. If wasm-pack does not have a private cache (fresh checkout) then the build will fail locally and on CI.

Two options I see: lock wasm-bindgen to the same version across the branches, or install wasm-bindgen in a per-repo location that wasm-pack will find. I do prefer per-repo bin directories for right versions of the tools for that repo, but I think that's a bigger job. locking wasm-bindgen to 0.2.87 on all branches is probably what we should do as that is the version wasm-pack 0.12.1 apparently want.

Any other ideas?